distributed/security.py

```diff
@@ -123,7 +123,8 @@ def get_connection_args(self, role):
         """
         d = {}
         tls = self.get_tls_config_for_role(role)
-        d['ssl_context'] = self._get_tls_context(tls, ssl.Purpose.SERVER_AUTH)
+        purpose = ssl.Purpose.SERVER_AUTH if hasattr(ssl, "Purpose") else None
+        d['ssl_context'] = self._get_tls_context(tls, purpose)
```

Comment:
I would rather have `_get_tls_context` return None (or, better, a dummy object that will raise an explicit error if one of its APIs is being used?) if the `ssl` module doesn't have the required APIs.